### DISCUSSION TITLE
Alpine js reusable components in separate html files and return components entirely with html and functionality within them.

### DISCUSSION THREAD

**Discussion by Shashank-Bhatt1** [alpinejs/alpine#3878] [1. Help]
https://stackoverflow.com/questions/77489208/get-html-files-path-in-other-package-inserted-html-in-alpine-js-component-based

I have above question related to embedding html in JavaScript module when using alpinejs in modular way so that i can leverage module based setup.

I am following alpine js in es6 modular way and need to have html returned from my own node package which i am developing right now. I need to return string containing `x-component` but path to html is not correct when using x-component syntax when it is rendered in browser. I need some way for adding path so my module works anywhere whether in Laravel or plain html project build with webpack,

As i already know alpine js doesn't provide the component based setup same as vue or react, however it will be good if i can get it working and return entire html from my module and also ability to keep html in separate html files from js logic.

Plus i want to return entire component with html so i can plug and play directly

Any help out there?

**Comment by ekwoka** [comment]
You would need to use an html loader and import the html itself as a module.

Here, you're doing runtime resolving of the path, which won't be the same after bundling, since nothing would tell it to move that indicated file to the same place as the bundled code.

(also, the appropriate way in JS to do path resolution is with `import.meta.resolve` but that also won't work here)

**Comment by Shashank-Bhatt1** [comment]
@ekwoka  Thank you for the response. 

Can you please give a short hint/example on how can i merge html with js to return component as i searched else where but could not found an optimal way. I need to keep html templates in my own node package only but need to get them working when someone requires/import my package. So i am not sure if there is cross everything way to manage paths.

I could for example write url="node_modukes/my-own-package/src/components/main/main.html" in x-component but that might not be the case for some other platforms,

It would be really helpful if you can share how to import html as a module to merge it within js to return the whole thing.

**Reply by ekwoka** [reply]
If you use vite (you should)

https://vitejs.dev/guide/assets.html#importing-asset-as-string

Otherwise you can look at how webpack can inline the html string for you to use or otherwise generate a module from it.

:)
